### PR TITLE
applications: machine_learning: Document nRF53 DK support

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces documentation related to support for nRF53 DK.

Jira: NCSDK-11647

### CODE REVIEW

File: applications/machine_learning/README.rst [hunk 1]
```diff
@@ -159,6 +182,7 @@ The application supports the following build types:
 * ``ZDebug`` -- Debug version of the application - can be used to verify if the application works correctly.
 * ``ZRelease`` -- Release version of the application - can be used to achieve better performance and reduce memory consumption.
 
+Not every board supports both mentioned build types.
```
Comment (emob-nordic) [inline applications/machine_learning/README.rst:185]: btw, why there is no Release version for nrf5340dk?

File: applications/machine_learning/README.rst [hunk 2]
```diff
@@ -196,6 +220,11 @@ By default, the following buttons are used by the application:
   * **Button 1** switches between data forwarding and running a machine learning model.
   * **Button 3** changes the signal generated by the simulated sensor.
 
+* nRF5340 Development Kit:
+
+  * **Button 1** switches between data forwarding and running a machine learning model.
+  * **Button 3** changes the signal generated by the simulated sensor.
+
```
Comment (emob-nordic) [inline applications/machine_learning/README.rst:227]: Couldn't this be squashed to previous bullet point? Like:
`nRF52840 and nRF5340 Development Kit:`

File: applications/machine_learning/README.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/machine_learning/README.rst:185]: I discussed this with Paweł and we decided to support only Debug variant here to limit number of configurations that we need to maintain.

File: applications/machine_learning/README.rst (hunk 2 above)
Comment (MarekPieta) [reply applications/machine_learning/README.rst:227]: This is also defined separately for Thingy boards

File: applications/machine_learning/README.rst [hunk 3]
```diff
@@ -284,11 +309,14 @@ The following configuration files can be defined for any supported board:
 Multi-image builds
 ------------------
 
-The Thingy:53 uses multi-image build with the following child images:
+The Thingy:53 and nRF53 Development Kit use multi-image build with the following child images:
 
 * MCUboot bootloader
 * Bluetooth HCI RPMsg
 
+Application-specific configuration for the mentioned child images can be defined in the board-specific directory in the :file:`applications/machine_learning/configuration/` directory.
```
Comment (wiba-nordic) [inline applications/machine_learning/README.rst:317]: Is this something that must be done? If that is the case, it's better like this:
```suggestion
Define the application-specific configuration for the mentioned child images in the board-specific directory in the :file:`applications/machine_learning/configuration/` directory.
```

File: applications/machine_learning/README.rst [hunk 4]
```diff
@@ -284,11 +309,14 @@ The following configuration files can be defined for any supported board:
...
+The Kconfig configuration files are named :file:`child_image_name_build_type`.
```
Comment (wiba-nordic) [inline applications/machine_learning/README.rst:318]:
```suggestion
The Kconfig configuration file names contain the name of the child image and the build type.
```

File: applications/machine_learning/README.rst (hunk 3 above)
Comment (MarekPieta) [reply applications/machine_learning/README.rst:317]: For MCUboot configuration it's optional. If it's not defined, default configuration is used

Comment (MarekPieta) [reply applications/machine_learning/README.rst:317]: I will use `You can define ...`

**Review by wiba-nordic — APPROVED** [review]
Looks good.

**Pull request merged**